### PR TITLE
Build can now produce proper MacOS `.app` and `.dmg` files

### PR DESCRIPTION
**Pull request by Flipp3rrr**
Built packages are placed into `bundles/`. The `freerct-<version>-macos-amd64.dmg` file is the one commonly distributed, but `FreeRCT.app` in `_CPack_Packages/amd64/Bundle/freerct-<version>-macos-amd64/FreeRCT.app` should also work by itself. I feel like I should note it's not exactly common practice to put the executable in `Resources/bin/<executable`, but it works and I believe it's not necessarily more complicated than placing it in `MacOS/<executable>` (which I couldn't figure out how to do).

### CODE REVIEW

**Comment by Flipp3rrr** [comment]
Side note: There's some more keys that aren't required that can be added to the `.plist` file, such as copyright information.

**Comment by Flipp3rrr** [comment]
I also forgot to mention the related issue, #260.

**Comment by Noordfrees** [comment]
LGTM :) 
I created a PR to your branch with a suggestion to integrate this into the CI: https://github.com/Flipp3rrr/FreeRCT/pull/1

**Comment by Flipp3rrr** [comment]
Merged the PR. I'm still working on a few things, since this still requires you to manually install dependencies.

**Comment by Newbytee** [comment]
Is this ready for review?

**Comment by Flipp3rrr** [comment]
> Is this ready for review?

@Newbytee sadly not yet, the code is still very messy and dependencies still have to be installed manually (any help with this would be appreciated).

**Comment by Flipp3rrr** [comment]
@Noordfrees I noticed there's a MacOS release [on the website](https://freerct.net/download) and there's a [Mastodon post](https://mastodon.technology/@FreeRCT/108051952595184894) about it, but the builds are broken.

**Comment by Flipp3rrr** [comment]
> @Noordfrees I noticed there's a MacOS release [on the website](https://freerct.net/download) and there's a [Mastodon post](https://mastodon.technology/@FreeRCT/108051952595184894) about it, but the builds are broken.

To be precise; it's broken if you download the build by itself, dependencies can be installed with homebrew to make it work but I don't think this should be labelled as a "release" if at all made available on the website.

File: cpack/PackageBundle.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+set(CPACK_BUNDLE_ICON "${CMAKE_SOURCE_DIR}/graphics/sprites/logo/logo.png")
+set(CPACK_BUNDLE_PLIST "${CMAKE_CURRENT_BINARY_DIR}/Info.plist.in")
+# https://cmake.org/cmake/help/latest/cpack_gen/bundle.html#variable:CPACK_BUNDLE_STARTUP_COMMAND
+set(CPACK_BUNDLE_STARTUP_COMMAND "${CMAKE_CURRENT_BINARY_DIR}/bin/freerct")
+set(CPACK_DMG_FORMAT "UDBZ")
+
+configure_file("${CMAKE_SOURCE_DIR}/packaging_data/osx/Info.plist.in" "${CMAKE_CURRENT_BINARY_DIR}/Info.plist.in")
+set(CPACK_BUNDLE_PLIST_SOURCE "${CMAKE_CURRENT_BINARY_DIR}/Info.plist.in")
+
+# Link libs, the wrong way (libpng sdl2 sdl2_ttf)
```
Comment (Flipp3rrr) [inline cpack/PackageBundle.cmake:13]: From this bit onwards the things done in this file are still completely broken, libraries are in fact not linked and the builds still don't work without manually installing them with homebrew.

**Comment by Noordfrees** [comment]
The homepage's daily builds updater accidentally also considered builds for PR branches in addition to master builds. I removed all the daily builds assets from the server now, tomorrow morning it should automatically provide the latest stable builds from master again. Thanks for noticing

**Comment by Flipp3rrr** [comment]
> The homepage's daily builds updater accidentally also considered builds for PR branches in addition to master builds. I removed all the daily builds assets from the server now, tomorrow morning it should automatically provide the latest stable builds from master again. Thanks for noticing

That's good to hear, thanks! As for the PR I'm not quite sure what to do, since I won't have time to fix the dependency issues any time soon.

**Pull request closed without merging**